hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java

```diff
@@ -0,0 +1,254 @@
+   */
+  private int deleteAllKeys(List<DeleteBlockGroupResult> results,
+      KeyManager manager) throws IOException {
+    Table<String, RepeatedOmKeyInfo> deletedTable =
+        manager.getMetadataManager().getDeletedTable();
+    DBStore store = manager.getMetadataManager().getStore();
+
+    // Put all keys to delete in a single transaction and call for delete.
+    int deletedCount = 0;
+    try (BatchOperation writeBatch = store.initBatchOperation()) {
+      for (DeleteBlockGroupResult result : results) {
+        if (result.isSuccess()) {
+          // Purge key from OM DB.
+          deletedTable.deleteWithBatch(writeBatch,
+              result.getObjectKey());
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("Key {} deleted from OM DB", result.getObjectKey());
+          }
+          deletedCount++;
+        }
+      }
+      // Write a single transaction for delete.
+      store.commitBatchOperation(writeBatch);
+    }
+    return deletedCount;
+  }
+
+  /**
+   * Submits PurgeKeys request for the keys whose blocks have been deleted
+   * by SCM.
+   * @param results DeleteBlockGroups returned by SCM.
+   */
+  public int submitPurgeKeysRequest(List<DeleteBlockGroupResult> results,
+      String snapTableKey) {
+    Map<Pair<String, String>, List<String>> purgeKeysMapPerBucket =
+        new HashMap<>();
+
+    // Put all keys to be purged in a list
+    int deletedCount = 0;
+    for (DeleteBlockGroupResult result : results) {
+      if (result.isSuccess()) {
+        // Add key to PurgeKeys list.
+        String deletedKey = result.getObjectKey();
+        // Parse Volume and BucketName
+        addToMap(purgeKeysMapPerBucket, deletedKey);
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Key {} set to be purged from OM DB", deletedKey);
+        }
+        deletedCount++;
+      }
+    }
+
+
+    OzoneManagerProtocolProtos.PurgeKeysRequest.Builder purgeKeysRequest =
+        OzoneManagerProtocolProtos.PurgeKeysRequest.newBuilder();
+
+    // Add keys to PurgeKeysRequest bucket wise.
+    for (Map.Entry<Pair<String, String>, List<String>> entry :
+        purgeKeysMapPerBucket.entrySet()) {
+      Pair<String, String> volumeBucketPair = entry.getKey();
+      OzoneManagerProtocolProtos.DeletedKeys deletedKeysInBucket =
+          OzoneManagerProtocolProtos.DeletedKeys.newBuilder()
+          .setVolumeName(volumeBucketPair.getLeft())
+          .setBucketName(volumeBucketPair.getRight())
+          .addAllKeys(entry.getValue())
+          .build();
+      purgeKeysRequest.addDeletedKeys(deletedKeysInBucket);
+    }
+
+    if (snapTableKey != null) {
+      purgeKeysRequest.setFromSnapshot(snapTableKey);
+    }
+
+    OzoneManagerProtocolProtos.OMRequest omRequest =
+        OzoneManagerProtocolProtos.OMRequest.newBuilder()
+        .setCmdType(OzoneManagerProtocolProtos.Type.PurgeKeys)
+        .setPurgeKeysRequest(purgeKeysRequest)
+        .setClientId(clientId.toString())
+        .build();
```

Comment:
```suggestion
    OMRequest omRequest =
        OMRequest.newBuilder()
        .setCmdType(Type.PurgeKeys)
        .setPurgeKeysRequest(purgeKeysRequest)
        .setClientId(clientId.toString())
        .build();
```

Reply:
And import these at the beginning just like existing KDS does:

```
import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.DeletedKeys;
import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMRequest;
import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.PurgeKeysRequest;
import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.Type;
```

It makes the code more readable.